### ISSUE TITLE
rates, counts, dead time. 

### ISSUE THREAD

**Issue by grburgess** [JamesPaynter/PyGRB#9]
The data loading class appears to be able to read both TTE and pre-binned counts data. However, I cannot find the functionality to handle the binning of the TTE data.

1. How would TTE data be handled?

2. I see that you convert rates to counts by multiplying with the duration of the time bin. However, this does not account for the instrumental dead time in the time bin and it results in non-integer values for the Poisson likelihood. It is better to get the counts directly, and multiply the rate function you are estimating by the exposure = bin duration - dead time. This maintains numerical stability and is mathematically proper. 

I'm not sure what info is in the BATSE rate files at the moment, but will investigate.

**Comment by grburgess** [comment]
here openjournals/joss-reviews#2536

**Comment by grburgess** [comment]
Here are some examples of doing this with GBM data:
https://github.com/threeML/threeML/blob/master/threeML/utils/data_builders/fermi/gbm_data.py#L16

It is broken up over several processes.

**Comment by JamesPaynter** [comment]
When you say the binning of tte data, do you mean you are looking for a class method to return the counts in bins?

1. I used the interpolate bins method to make a binary array in units of 2 us and then used that for analysis.

2. I had thought that that is how they had made their rates, because if you multiply the rates by the binwidths you get an array of ~ integers within half a percent. I thought that difference was a due to floating point arithmetic, but looking at a graph of a burst and roundoff error, there seems to be some correlation:
![roundoff](https://user-images.githubusercontent.com/45152710/89493982-916e2f00-d7f8-11ea-814b-0732dab530ac.png)

They have summed over triggered detectors to make the light-curves, I am not sure that counts are available for most bursts. There is ~ 2 sec of tte data available for all bursts but only fully contains short and moderately bright bursts.

"DISCSC consist of 64 ms discriminator data summed from each of the LAD's that were triggered by the burst."
https://heasarc.gsfc.nasa.gov/docs/cgro/nra/appendix_g.html#V.%20BATSE%20GUEST%20INVESTIGATOR%20PROGRAM

So I think in most cases that is the best data available. Would it be better to take it back to counts by multiplying by (binwidths - deadtime)?

**Comment by grburgess** [comment]
What is the availability for TTE? Is it there for all GRBs? If so, I would focus on binning the TTE for light curve analysis.

**Comment by JamesPaynter** [comment]
It is available for all, but because of hardware limitations it only catches 2 seconds or a set amount of photons, whichever comes first. So in general the tte data is not that useful for long gamma-ray bursts, but a good percentage of short gamma-ray bursts are completely resolved in tte. Although bright ones fill up the buffer faster, so they are also not completely resolved in tte.
eg.
![batse2611](https://user-images.githubusercontent.com/45152710/89513258-aefec100-d817-11ea-9838-1f09b25cb705.PNG)

tl;dr A short amount of tte data (~2s) is available for all bursts, but in most cases it is not useful 😞

**Comment by grburgess** [comment]
That does present a problem. With that in mind, it is important to make sure the likelihood is correct with the data. If there is no counts extension in the PHA files, then it will be difficult to do this correctly. Is there an exposure keyword?

**Comment by JamesPaynter** [comment]
I'm pretty sure the short answer is no, 
(it is in all the file types I've checked so far, but might be under another keyword as exposure for BATSE seems to mean sky exposure).
I'm still double checking what the long answer is.

**Comment by JamesPaynter** [comment]
I think unfortunately the long answer is also no.
I have been conversing with someone through the HEASARC help desk for the past couple of days but it is still not clear to me how counts at the detector(s) have been converted to the 'RATES' keyword in bfits files.
It may have been something that was done aboard the satellite due to limited telemetry bandwidth.

**Comment by grburgess** [comment]
I was worried this would be true. I will talk with some people I know in BATSE and see if I can figure out a solution.

**Comment by JamesPaynter** [comment]
That would be great! Thank you

**Comment by grburgess** [comment]
"re: detector summing, is it then appropriate to have a radiation field which is convolved / folded with the triggered detector response matrices to then compare to observed counts in the Poisson likelihood?
This would work for the tte data.

I'm still not sure of a way forward using the main available datatypes (eg discsc 64ms binned data) as my understanding is they come pre-summed over triggered detectors (prebinned on the satellite). There is not tte data available to cover the entire time history of ~80-90% of BATSE GRBs, only lower resolution eg. discsc data.
"

If you can find a way to fold a model through the data, this would work very well. 

Yes, I couldn't find enough information on this as well. While it is not completely appropriate, perhaps it is best to work with the prebinned data but warn the user of the caveats.
